### ISSUE TITLE
Unable to initialize shiny driver on rstudio server.

### ISSUE THREAD

**Issue by dougsanders79** [rstudio/shinytest#104]
Hello,

I'm attempting to run recordTest() on RStudio Server but I get the following error:

```
> library(shiny)
> library(shinytest)
> recordTest()

**Error in sd_startShiny(self, private, path) : 
Cannot find shiny port number. Error: Running application in test mode.Loading required package: shiny**
```
_SessionInfo:_
```
RStudio Server: 1.0.136
R: 3.4.0
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Red Hat Enterprise Linux Server release 6.6 (Santiago)
```
I'm guessing this is because ShinyDriver is unable to open a new port on the server? The Rstudio Server instance does not run as the root user. 

recordTest() works perfectly okay on my Mac.

Is there a way to have recordTest() open a port locally on my laptop and have it run there, much like how RStudio Server creates a local port for the Shiny app to run?

Thanks,
Doug

**Comment by KZARCA** [comment]
Hi,
I'm facing a similar issue, did you try with another simpler app?

**Comment by dougsanders79** [comment]
We did try a simpler app as I recall.  I did manage to get it to work
running locally (i.e on my laptop) it was just the Rstudio server it
wouldn't work on.  t've put it on a back burner for now and may try to work
it again in some weeks.

One idea which I thought about later was to try to force the shiny app to
run local window using:

 options = list(height = height, width = width, shiny.launch.browser =
.rs.invokeShinyWindowViewer))

but don't have time to go back and try that right now (admittedly it really
shouldn't take long to put that line in, maybe try a few different things
for shiny.launch.browser.  But it may be that the testing somehow overrides
it anyway and tries to open the window instance on the server.

DS
